src/shared/mkfs-util.c

```diff
@@ -432,6 +432,15 @@ int make_filesystem(
                 if (quiet && strv_extend(&argv, "-q") < 0)
                         return log_oom();
 
+                if (sector_size > 0)
+                        FOREACH_STRING(s, "MKE2FS_DEVICE_SECTSIZE", "MKE2FS_DEVICE_PHYS_SECTSIZE") {
```

Comment:
hmm, what precisely is the difference betwen a "physical" and a "logical sector size"? what does mke2fs precisely do with this?

it's so weird, hat mke2fs wants this as env var rather than regular switch

Reply:
what about the other file system types?

Reply:
mke2fs uses this to figure out the default block size or perform some additional checks against the block size if explicitly configured. I figure it makes sense to set these explicitly as otherwise mke2fs will try to figure them out from a device and we don't want that.

We already configure the sector size for the other file system types where possible.

Reply:
still, what's the difference between the two sector sizes? i don't get it?

Reply:
This explains it somewhat: https://superuser.com/questions/982680/whats-the-point-of-hard-drives-reporting-their-physical-sector-size

I guess we could opt to not set the physical sector size environment variable, but tbf I want to avoid mkfs trying to look it up from the host disk or stuff like that.

Reply:
are you sure it's about that? in the context of mke2fs? I kinda sense that different areas use the two words differently, which makes it so utterly confusing

Reply:
I opened https://github.com/systemd/systemd/pull/28839 to not touch the physical sector size